README.md

```diff
@@ -7,3 +7,522 @@ This repository contains submodules of all the repositories used for skywire dep
 * [skywire-utilities](https://github.com/skycoin/skywire-utilities)
 * [skywire-services](https://github.com/skycoin/skywire-services)
 * [skycoin-service-discovery](https://github.com/skycoin/skycoin-service-discovery)
+
+
+## Code Checks & Tests
+
+Dependencies: [go](https://go.dev/) [goimports](https://pkg.go.dev/golang.org/x/tools/cmd/goimports) & [goimports-reviser](https://github.com/incu6us/goimports-reviser)
+
+For any given submodule repository, included is a Makefile with directives such as `format` and `check`.
+`make format check` is currently used by the CI to check pull requests.
+
+`make format check` executes a variation of the following commands
+```
+go mod tidy -v
+[[ -d pkg ]] && goimports -w -local github.com/skycoin/$(basename $(pwd)) ./pkg
+[[ -d cmd ]] && goimports -w -local github.com/skycoin/$(basename $(pwd)) ./cmd
+[[ -d internal ]] && goimports -w -local github.com/skycoin/$(basename $(pwd)) ./internal
+find . -type f -name '*.go' -not -path "./.git/*" -not -path "./vendor/*"  -exec goimports-reviser -project-name github.com/skycoin/$(basename $(pwd)) {} \;
+go clean -testcache &>/dev/null
+[[ -d internal ]] && go test  -cover -timeout=5m -mod=vendor ./internal/...
+[[ -d pkg ]] && go test -cover -timeout=5m -mod=vendor ./pkg/...
+```
+
+The flags used for `go test` may vary from repo to repo
+
+## Building
+
+Dependencies: [go](https://go.dev/)
+
+For any given submodule repository (except skywire-utilities) included is a Makefile with a `build` directive
+
+`make build` executes `go build` for all the binaries that can be built from .go files in `./cmd` subdirectories, for example
+
+```
+	go build ./cmd/skywire-visor
+```
+
+Note that some binaries may have gcc / libc6 dependency unless statically compiled with musl.
+
+## Runtime Dependencies by Service
+
+### Redis
+* address-resolver
+* network-monitor
+* dmsg-discovery
+
+### Postgres
+* transport-discovery
+* route-finder
+* service-discovery
+
+#### DB Setup
+
+Some services use postgresql.
```

Comment:
Should be more specific and include the dbs that actually need to be created. So go through the dbs that should be created after installing. The names can change, but we can tell administrators that they need to create four dbs with stand-in names:

Example of what I mean: 

```
Create the following dbs :

- service-discovery
- transport-discovery
- etc

```

We should also mention that we may need to create a user for Skywire deployment and grant it permissions on the dbs we create.

Reply:
> Currently the documentation is not copy/pastable for reasons I go into in the specific comments. We need to work on that a bit.
> 
> Ports are not explicitly mentioned. For most of the ports, its sufficient to say that they can be variable and are not magic. As long as the adminstrator forwards the incoming http/https ports to the ports set during runtime via arguments they can be anything. The only cases where that is not the case is the dmsg server as far as I know.
> 
> Apart from that, this is a good start and we should be able to get this done tomorrow. Well done...

Where are the ports not explicitly mentioned?
Every service which can have a port set has the default port shown in the invocation
```
dmsg-discovery --addr ":9090" --redis "redis://localhost:6379" --sk $(tail -n1 dmsgd-config.json)
```
...
```
--addr ":9090"
```
...
```
address-resolver --addr ":9093" --redis "redis://localhost:6379" --dmsg-disc "http://dmsgd.skywire.skycoin.com" --sk $(tail -n1 ar-config.json)
```
...
```
 --addr ":9093"
```
...
```
PG_USER="postgres" PG_DATABASE="route-finder" PG_PASSWORD="" route-finder  --addr ":9092" --dmsg-disc "http://dmsgd.skywire.skycoin.com" --sk $(tail -n1 rf-config.json)
```
...
```
 --addr ":9092"
```
...
```
PG_USER="postgres" PG_DATABASE="tpd" PG_PASSWORD="" transport-discovery  --addr ":9091" --dmsg-disc "http://dmsgd.skywire.skycoin.com" --sk $(tail -n1 tpd-config.json)
```
...
```
 --addr ":9091" 
```
...
```
PG_USER="postgres" PG_DATABASE="sd" PG_PASSWORD="" service-discovery  --addr ":9098" --dmsg-disc "http://dmsgd.skywire.skycoin.com" --sk $(tail -n1 sd-config.json)
```
...
```
--addr ":9098"
```
...
```
network-monitor -c nm-config.json -a ":9080"
```
...
```
-a ":9080"
```
...
and for the visor the config is shown, as well as the default ports

Note the default ports:

:7777
:3435
:5505
:8000
:8001

Note: tcp, udp, and http ports in the config will always have a colon and will never be low ports